tests/integration/src/test/resources/org/mule/test/integration/client/client-dispatch-catch-exception-flow.xml

```diff
@@ -0,0 +1,71 @@
+        <logger message="inner flow called..." level="INFO"/>
+        <custom-processor class="org.mule.test.integration.client.MuleClientDispatchExceptionHandlingTestCase$InnerFlowCountDownProcessor"/>
+    </flow>
+
+    <flow name="catchExceptionJavaComponentUsingVM">
+        <vm:inbound-endpoint exchange-pattern="request-response" path="vmcatchexception"/>
+        <set-payload value="empty"/>
+        <component class="org.mule.test.integration.client.MuleClientDispatchExceptionHandlingTestCase$DispatchInnerFlowThrowExceptionJavaComponent"/>
+        <set-payload value="Sales"/>
+        <catch-exception-strategy>
+            <set-payload value="Business Error!"/>
+            <custom-processor class="org.mule.test.integration.client.MuleClientDispatchExceptionHandlingTestCase$ExecutionCountDownProcessor"/>
+        </catch-exception-strategy>
+    </flow>
+
+    <flow name="catchExceptionUsingVMConnector">
+        <vm:inbound-endpoint exchange-pattern="request-response" path="vmconnector"/>
+        <set-payload value="empty"/>
+        <vm:outbound-endpoint exchange-pattern="one-way" path="vminnertest"/>
+        <component class="org.mule.test.integration.client.MuleClientDispatchExceptionHandlingTestCase$ThrowException"/>
+        <set-payload value="Sales"/>
+        <catch-exception-strategy>
+            <set-payload value="Business Error!"/>
+            <custom-processor class="org.mule.test.integration.client.MuleClientDispatchExceptionHandlingTestCase$ExecutionCountDownProcessor"/>
+        </catch-exception-strategy>
+    </flow>
+
+    <spring:beans>
+        <spring:bean id="myProcessor"
+                     class="org.mule.test.integration.client.MuleClientDispatchExceptionHandlingTestCase$DispatchInnerFlowThrowExceptionMessageProcessor"/>
+    </spring:beans>
+
+    <flow name="catchExceptionMessageProcessorUsingVM">
+        <vm:inbound-endpoint exchange-pattern="request-response" path="vmcatchexceptionfromprocessor"/>
+        <set-payload value="empty"/>
+        <processor ref="myProcessor"/>
+        <set-payload value="Sales"/>
+        <catch-exception-strategy>
+            <set-payload value="Business Error!"/>
+            <custom-processor class="org.mule.test.integration.client.MuleClientDispatchExceptionHandlingTestCase$ExecutionCountDownProcessor"/>
+        </catch-exception-strategy>
+    </flow>
+
+    <flow name="catchExceptionJavaComponentUsingHTTP">
+        <http:listener config-ref="HTTP_Listener_Configuration" path="/test" />
+        <set-payload value="empty"/>
+        <component class="org.mule.test.integration.client.MuleClientDispatchExceptionHandlingTestCase$DispatchInnerFlowThrowExceptionJavaComponent"/>
+        <set-payload value="Sales"/>
+        <catch-exception-strategy>
+            <set-payload value="Business Error!"/>
+            <custom-processor class="org.mule.test.integration.client.MuleClientDispatchExceptionHandlingTestCase$ExecutionCountDownProcessor"/>
```

Comment:
Consider using FlowAsserts.verify("flowVarsFlowUsingVM") and test:assert/ processors as it make the test much more readable/direct.